zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbAuthorizationState.java

```diff
@@ -162,6 +162,38 @@ public void deleteAuthorizationsByOwnerTypeAndIdPrefix(
         });
   }
 
+  @Override
+  public void update(final long authorizationKey, final AuthorizationRecord authorization) {
+    this.authorizationKey.wrapLong(authorizationKey);
+    final var persistedAuthorization =
+        authorizationByAuthorizationKeyColumnFamily.get(this.authorizationKey);
+
+    if (persistedAuthorization == null) {
+      // if there are no persisted authorization, we should not update the authorization
+      return;
+    }
+
+    // remove the old permissions
+    persistedAuthorization
+        .getPermissions()
+        .forEach(
+            permissionType -> {
+              removePermission(
+                  persistedAuthorization.getOwnerType(),
+                  persistedAuthorization.getOwnerId(),
+                  persistedAuthorization.getResourceType(),
+                  permissionType,
+                  Set.of(persistedAuthorization.getResourceId()));
+            });
+
+    // delete the old authorization record
+    authorizationByAuthorizationKeyColumnFamily.deleteExisting(this.authorizationKey);
+    permissionKeyByAuthorizationKeyColumnFamily.deleteExisting(this.authorizationKey);
```

Comment:
💭 This CF might not be needed anymore, but I'll wait until also the delete method is implemented to properly evaluate this

Reply:
❓ Do we still need the `permissionsColumnFamily`? Don't we have the permissions inside of the `PersistedAuthorization` as well?

Reply:
We have the permission related only to one resourceId, the PERMISSION cf store all the permission for a specific owner and resource type